### ISSUE TITLE
Native AES fails while loading the default backend

### ISSUE THREAD

**Issue by klemens** [kalkin/gajim-omemo#99]
I just updated to the new 0.8.0 version, which tries to use native crypto if possible. This however leads to the following exception when receiving a encrypted message:

```
05.08.2016 18:33:39 (E) gajim.c.ged Error while running an even handler: <bound method OmemoPlugin.message_received of <gajim-omemo.OmemoPlugin object at 0x7f5916e3ebd0>>
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/gajim/common/ged.py", line 93, in raise_event
    if handler(*args, **kwargs):
  File "/usr/lib/python2.7/site-packages/gajim/plugins/helpers.py", line 107, in wrapper
    result = f(*args, **kwargs)
  File "/home/klemens/.local/share/gajim/plugins/gajim-omemo/__init__.py", line 233, in message_received
    plaintext = state.decrypt_msg(msg_dict)
  File "/home/klemens/.local/share/gajim/plugins/gajim-omemo/omemo/state.py", line 215, in decrypt_msg
    result = unicode(decrypt(key, iv, payload))
  File "/home/klemens/.local/share/gajim/plugins/gajim-omemo/omemo/aes_gcm.py", line 39, in decrypt
    plaintext = aes_decrypt(key, iv, ciphertext).decode('utf-8')
  File "/home/klemens/.local/share/gajim/plugins/gajim-omemo/omemo/aes_gcm_native.py", line 43, in aes_decrypt
    _backend = default_backend()
  File "/usr/lib/python2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 35, in default_backend
    _default_backend = MultiBackend(_available_backends())
  File "/usr/lib/python2.7/site-packages/cryptography/hazmat/backends/__init__.py", line 21, in _available_backends
    for ep in pkg_resources.iter_entry_points(
  File "/usr/lib/python2.7/site-packages/gajim/common/demandimport.py", line 86, in __getattribute__
    self._load()
  File "/usr/lib/python2.7/site-packages/gajim/common/demandimport.py", line 58, in _load
    mod = _origimport(head, globals, locals)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2958, in <module>
    @_call_aside
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2944, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2986, in _initialize_master_working_set
    dist.activate(replace=False)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2518, in activate
    fixup_namespace_packages(self.location)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2110, in fixup_namespace_packages
    _imp.acquire_lock()
  File "/usr/lib/python2.7/site-packages/gajim/common/demandimport.py", line 86, in __getattribute__
    self._load()
  File "/usr/lib/python2.7/site-packages/gajim/common/demandimport.py", line 58, in _load
    mod = _origimport(head, globals, locals)
ImportError: No module named _imp
```

Manually executing the `default_backend` function in a python2 shell works without problems:

```
>>> from cryptography.hazmat.backends import default_backend
>>> default_backend()
<cryptography.hazmat.backends.multibackend.MultiBackend object at 0x7f351a0aee10>
```

Reverting to using the built-in encryption fixes the exception and the messages get decrypted as before.

I am running gajim 0.16.5 and cryptography 1.4 using the default packages on arch.

**Comment by lovetox** [comment]
ah i know this error, has nothing to do with the crypto

its the package loader module from python thats not playing well with demandimport

i forgot to add this to the ignore list

i upload a fast fix in another branch could you test it?

**Comment by lovetox** [comment]
https://github.com/lovetox/gajim-omemo

could you try from this repo please

**Comment by klemens** [comment]
https://github.com/lovetox/gajim-omemo/commit/1ac0b855aa6bdd634de83d3a5c781d1edca3ad16 indeed fixes the problem. Thanks for your fast response!

**Comment by lovetox** [comment]
changes are merged, v0.8.1 will soon be on gajim.
